### PR TITLE
AP_Scripting: add winch binding and example script

### PR DESCRIPTION
**Pull request by rmackay9**
This adds bindings and documentation allowing Lua scripts to control a winch.

An example script is also included that raises or lowers the winch at configurable rates using an auxiliary switch or a servo **output** (see explanation below).

I require this functionality for an [upcoming search and rescue event](https://japan-innovation-challenge.com/en/) which stipulates the pilot must remotely operate the vehicle from >4km away.  My team will use the 3G/LTE mobile phone network, Mission Planner and a Real Flight joystick to control the vehicle.  The issue is that, like many joysticks, the RealFlight joystick only has 5 channels with full range meaning we do not have an RC channel free to set the winch deploy/retract rate.

The example script gets around this issue by allowing the user to set the desired deploy and retract speeds in WINCH_RATE_UP/DN parameters.  If they have a 3 position auxiliary switch they can then choose whether to retract, hold or deploy the winch line.  Where it gets weird is if absolutely no RC switch is available the user can setup a servo **output** to be used in place of the RC input.  Mission Planner's Servo/Relay tab can then be used to set that servo output which is then consumed as an input like an aux function switch.

Of course we should consider whether we have enough winch users to justify the additional flash cost of the winch bindings.

This has been tested in SITL

### CODE REVIEW

File: libraries/AP_Scripting/examples/winch-test.lua [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+assert(param:add_param(PARAM_TABLE_KEY, 3, "SRV_SRC_FN", -1), "could not add WINCH_SRV_SRC_FN param")
+
+local winch_rate_up = Parameter()
+assert(winch_rate_up:init("WINCH_RATE_UP"), "could not find WINCH_RATE_UP param")
+
+local winch_rate_dn = Parameter()
+assert(winch_rate_dn:init("WINCH_RATE_DN"), "could not find WINCH_RATE_DN param")
+
+local winch_srv_src_fn = Parameter()
+assert(winch_srv_src_fn:init("WINCH_SRV_SRC_FN"), "could not find WINCH_SRV_SRC_FN param")
```
Comment (IamPete1) [inline libraries/AP_Scripting/examples/winch-test.lua:34]: On master you can just:
```suggestion
local winch_rate_up = Parameter"WINCH_RATE_UP"()
local winch_rate_dn = Parameter("WINCH_RATE_DN")
local winch_srv_src_fn = Parameter("WINCH_SRV_SRC_FN")
```

File: libraries/AP_Scripting/generator/description/bindings.desc [hunk 2]
```diff
@@ -551,6 +551,14 @@ singleton AP_Mount method set_rate_target void uint8_t 0 UINT8_MAX float'skip_ch
 singleton AP_Mount method set_roi_target void uint8_t 0 UINT8_MAX Location
 singleton AP_Mount method get_attitude_euler boolean uint8_t 0 UINT8_MAX float'Null float'Null float'Null
 
+include AP_Winch/AP_Winch.h
+singleton AP_Winch rename winch
```
Comment (IamPete1) [inline libraries/AP_Scripting/generator/description/bindings.desc:555]:
```suggestion
include AP_Winch/AP_Winch.h depends APM_BUILD_COPTER_OR_HELI
singleton AP_Winch rename winch depends APM_BUILD_COPTER_OR_HEL
```

Comment (IamPete1) [reply libraries/AP_Scripting/generator/description/bindings.desc:555]: Oppps, wrong syntax, I think this should work:
```suggestion
include AP_Winch/AP_Winch.h depends APM_BUILD_COPTER_OR_HELI
singleton AP_Winch depends APM_BUILD_COPTER_OR_HELI
singleton AP_Winch rename winch
```

Comment (rmackay9) [reply libraries/AP_Scripting/generator/description/bindings.desc]: Ah, nevermind, I've found what works. txs!

File: libraries/AP_Scripting/examples/winch-test.lua (hunk 1 above)
Comment (rmackay9) [reply libraries/AP_Scripting/examples/winch-test.lua:34]: cool, txs.  fixed.

**Pull request merged**